From: Jiri Slaby <jslaby@suse.cz>
Subject: kABI: restore icmp_send
Patch-mainline: never, kabi
References: kabi

In networking-stable-19_03_07, upstream commit
9ef6b42ad6fd7929dd1b6092cb02014e382c6a91 (net: Add __icmp_send helper.)
renamed icmp_send function to __icmp_send and introduced an inline
helper icmp_send.  It made the kABI checker to complain.

Change the inline into a proper function and export it again.

Signed-off-by: Jiri Slaby <jslaby@suse.cz>
---
 include/net/icmp.h |    7 +------
 net/ipv4/icmp.c    |    5 +++++
 2 files changed, 6 insertions(+), 6 deletions(-)

--- a/include/net/icmp.h
+++ b/include/net/icmp.h
@@ -22,7 +22,6 @@
 
 #include <net/inet_sock.h>
 #include <net/snmp.h>
-#include <net/ip.h>
 
 struct icmp_err {
   int		errno;
@@ -42,11 +41,7 @@ struct net;
 
 void __icmp_send(struct sk_buff *skb_in, int type, int code, __be32 info,
 		 const struct ip_options *opt);
-static inline void icmp_send(struct sk_buff *skb_in, int type, int code, __be32 info)
-{
-	__icmp_send(skb_in, type, code, info, &IPCB(skb_in)->opt);
-}
-
+void icmp_send(struct sk_buff *skb_in, int type, int code, __be32 info);
 int icmp_rcv(struct sk_buff *skb);
 void icmp_err(struct sk_buff *skb, u32 info);
 int icmp_init(void);
--- a/net/ipv4/icmp.c
+++ b/net/ipv4/icmp.c
@@ -750,6 +750,11 @@ out:;
 }
 EXPORT_SYMBOL(__icmp_send);
 
+void icmp_send(struct sk_buff *skb_in, int type, int code, __be32 info)
+{
+       __icmp_send(skb_in, type, code, info, &IPCB(skb_in)->opt);
+}
+EXPORT_SYMBOL(icmp_send);
 
 static void icmp_socket_deliver(struct sk_buff *skb, u32 info)
 {
